### ISSUE TITLE
"Omit Message Scope" makes the TTS engine read the actual HTML of the message

### ISSUE THREAD

**Issue by Popkornium18** [mumble-voip/mumble#4309]
**Describe the bug**
The new option "Omit Message Scope" introduced in #3677 makes the TTS engine read the actual HTML.

**Steps to Reproduce**
1. Set Omit Message Scope in settings
2. Let someone else send you a message
3. Hear HTML-Tags
4. Render HTML in your mind

**Expected behavior**
The same message should be read, just without the scope.

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Version: Mumble compiled from master

**Comment by Krzmbrzl** [comment]
My guess would be that right here
https://github.com/mumble-voip/mumble/blob/a20d16777ef5f9b29e58c406bc1847d484475082/src/mumble/Messages.cpp#L901-L907
one shouldn't use the plain text as the TTS override but rather a HTML-parsed version of the message :thinking:

**Comment by Popkornium18** [comment]
I think it should just be converted to plain in `Log::log` as well.

https://github.com/mumble-voip/mumble/blob/a20d16777ef5f9b29e58c406bc1847d484475082/src/mumble/Log.cpp#L507

**Comment by Popkornium18** [comment]
Do you want to fix this or should I do it?

**Comment by Krzmbrzl** [comment]
Yeah I agree that it should be fixed before the function call.

> Do you want to fix this or should I do it?

Please go ahead - I have enough other stuff to do, so every PR I don't have to create myself is a good PR :D

**Comment by Popkornium18** [comment]
Ok, theres even more broken here... If you omit both scope and author tts reads "Colon this is the message" because `%2%1: %1` gets formatted as `: this is the message`.

I dont really understand this translation business in all detail. What exactly gets translated when you call `tr("%2%1: %3")`? Just the colon? If thats the case, we might as well just concatenate channel source and message because the colon shouldn't be read by TTS ever.

**Comment by Popkornium18** [comment]
Btw the issue is that `name` contains the HTML formatted message. It should be `pSrc->qsSname`. I definitely can see why this got mixed up.

But other than that, the way `overrideTTS` is generated looks overly complicated to me, so I'd like to simplify it, if possible.

**Comment by Krzmbrzl** [comment]
> What exactly gets translated when you call tr("%2%1: %3")?

Literally that string. So in this case the translation would look  the same in all languages as it doesn't contain any translatable String. Thus the translation is pretty useless anyways...

>  If you omit both scope and author tts reads "Colon this is the message" because %2%1: %1 gets formatted as : this is the message.

Then I guess the existence of these should be checked and depending on that the String should be assembled with or without that colon :thinking: 

> Btw the issue is that name contains the HTML formatted message. It should be pSrc->qsSname. I definitely can see how this got mixed up.

:facepalm: 
Yeah that's definitely confusing...

> But other than that, the way overrideTTS is generated looks overly complicated to me, so I'd like to simplify it, if possible.

Sure go ahead. Simpler code is almost always better (as long as it works of course) :)
